okio/src/main/java/okio/RealBufferedSource.java

```diff
@@ -233,6 +233,16 @@ public RealBufferedSource(Source source) {
     return buffer.readLongLe();
   }
 
+  @Override public long readDecimalLong() throws IOException {
+    request(20);
```

Comment:
This isn't particularly good. Imagine you're reading from a blocking stream that offers the following data immediately:

```
1 abc
```

Subsequent data isn't available immediately. But you don't need it; you only need 2 chars. With the `request(20)` call, you'll block needlessly. For interactive protocols, that unnecessary blocking could even cause you to hang.